content/200-concepts/100-components/01-prisma-schema/10-postgresql-extensions.mdx

```diff
@@ -0,0 +1,108 @@
+
+```prisma file=schema.prisma
+datasource db {
+  provider = "postgresql"
+  url      = env("TEST_DATABASE_URL")
+  extensions = [uuidOssp(map: "uuid-ossp")]
+}
+```
+
+## How to introspect PostgreSQL extensions
+
+To [introspect](/concepts/components/introspection) PostgreSQL extensions currently installed in your database and add relevant extensions to your Prisma schema, run `npx prisma db pull`.
+
+Many PostgreSQL extensions are not relevant to the Prisma schema. For example, some extensions are intended for database administration tasks that do not change the schema. If all these extensions were included, the list of extensions would be very long. To avoid this, Prisma maintains an allowlist of known relevant extensions. The current allowlist is the following:
+
+- [`citext`](https://www.postgresql.org/docs/current/citext.html): provides a case-insensitive character string type, `citext`
+- [`pgcrypto`](https://www.postgresql.org/docs/current/pgcrypto.html): provides cryptographic functions
+- [`uuid-ossp`](https://www.postgresql.org/docs/current/uuid-ossp.html): provides functions to generate universally unique identifiers (UUIDs)
+- [`postgis`](https://postgis.net/): adds GIS (Geographic Information Systems) support
+
+Extensions are introspected as follows:
+
+- The first time you introspect, all database extensions that are on the allowlist are added to your Prisma schema
+- When you re-introspect, the behavior depends on whether the extension is on the allowlist or not.
+  - Extensions on the allowlist:
+    - are **added** to your Prisma schema if they are in the database but not in the Prisma schema
+    - are **kept** in your Prisma schema if they are in the Prisma schema and in the database
+    - are **removed** from your Prisma schema if they are in the Prisma schema but not the database
+  - Extensions not on the allowlist:
+    - are **kept** in your Prisma schema if they are in the Prisma schema and in the database
+    - are **removed** from your Prisma schema if they are in the Prisma schema but not the database
+
+The `version` argument will not be added to the Prisma schema when you introspect.
+
+## How to migrate PostgreSQL extensions
+
+You can update your list of PostgreSQL extensions in your Prisma schema and apply the changes to your database with [Prisma Migrate](/concepts/components/prisma-migrate).
+
+This works in a similar way to migration of other elements of your Prisma schema, such as models or fields. However, there are the following differences:
+
+- If you remove an extension from your schema but it is still installed on your database, Prisma Migrate will not uninstall it from the database.
+- If you add a new extension to your schema, it will only be installed if it does not already exist in the database, because the extension may already have been created manually.
```

Comment:
```suggestion
- If you add a new extension to your schema, it will only be installed if it does not already exist in the database, because the extension may already have been created manually.
- If you remove the `version` or `schema` arguments from the extension definition, it has no effect to the extensions in the database in the following migrations.
```